services/src/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/stats/MflStatSpec.groovy

```diff
@@ -0,0 +1,134 @@
+        def waitTime = 120
+        def waitInterval = 10
+        def initStat
+        Wrappers.wait(waitTime, waitInterval) {
+            initStat = otsdb.query(startTime, metric, tags).dps
+            assert initStat.size() >= 1
+        }
+
+        when: "Set management controller only on the src switch"
+        lockKeeper.setController(srcSwitch.dpId, managementController)
+
+        and: "Generate traffic on the given flow"
+        exam.setResources(traffExam.startExam(exam, true))
+        assert traffExam.waitExam(exam).hasTraffic()
+
+        then: "Stat on the src switch should be collected because management controller is set"
+        def statFromMgmtController
+        Wrappers.wait(waitTime, waitInterval) {
+            statFromMgmtController = otsdb.query(startTime, metric, tags).dps
+            assert statFromMgmtController.size() > initStat.size()
+            assert statFromMgmtController[statFromMgmtController.keySet()[-2]] <
+                    statFromMgmtController[statFromMgmtController.keySet()[-1]]
+        }
+
+        when: "Set statistic controller only on the src switch"
+        lockKeeper.setController(srcSwitch.dpId, statController)
+
+        and: "Generate traffic on the given flow"
+        exam.setResources(traffExam.startExam(exam, true))
+        assert traffExam.waitExam(exam).hasTraffic()
+
+        then: "Stat on the src switch should be collected because statistic controller is set"
+        def statFromStatsController
+        Wrappers.wait(waitTime, waitInterval) {
+            statFromStatsController = otsdb.query(startTime, metric, tags).dps
+            assert statFromStatsController.size() > statFromMgmtController.size()
+            assert statFromStatsController[statFromStatsController.keySet()[-2]] <
+                    statFromStatsController[statFromStatsController.keySet()[-1]]
+        }
+
+        when: "Disconnect the src switch from the management and statistic controllers"
+        lockKeeper.knockoutSwitch(srcSwitch.dpId)
+        Wrappers.wait(WAIT_OFFSET) { assert !(srcSwitch.dpId in northbound.getActiveSwitches()*.switchId) }
+
+        and: "Generate traffic on the given flow"
+        exam.setResources(traffExam.startExam(exam, true))
+        assert traffExam.waitExam(exam).hasTraffic()
+
+        then: "Stat on the src switch should not be collected because it is disconnected from controllers"
+        double interval = waitTime * 0.2
+        def statAfterDeletingControllers
+        Wrappers.timedLoop(waitTime) {
+            statAfterDeletingControllers = otsdb.query(startTime, metric, tags).dps
+            assert statAfterDeletingControllers.size() == statFromStatsController.size()
+            sleep((interval * 1000).toLong())
+        }
+
+        when: "Restore default controllers on the src switches"
+        lockKeeper.reviveSwitch(srcSwitch.dpId)
+        Wrappers.wait(discoveryTimeout * 1.5 + WAIT_OFFSET) {
```

Comment:
1. why `*1.5`?
2. why `discoveryTimeout`? It should be `discoveryInterval`

Reply:
fixed